--- FormatDesc.java
+++ FormatDesc.java
@@ -239,9 +239,12 @@
 			buf.append(Character.isUpperCase(conversion) ? "INF" : "inf");
 			if (leftAdjust) pad(buf, ' ', effectiveWidth - 3);
 		} else {
+			/*
 			if (useOracleFormatting) {
 				try {
+			*/
 					formatWithOracle(buf, number, effectiveWidth);
+			/*
 					return;
 				} catch (LinkageError ignored) {
 					useOracleFormatting = false;
@@ -261,6 +264,7 @@
 			format.append((char) conversion);
 
 			buf.append(String.format(format.toString(), number));
+			*/
 		}
 	}
 
