--- original\Cobalt\src\main\java\org\squiddev\cobalt\lib\FormatDesc.java
+++ src\main\java\org\squiddev\cobalt\lib\FormatDesc.java
@@ -215,9 +215,12 @@
 			buf.append(Character.isUpperCase(conversion) ? "INF" : "inf");
 			if (leftAdjust) pad(buf, ' ', effectiveWidth - 3);
 		} else {
+			/*
 			if (useOracleFormatting) {
 				try {
+			*/
 					formatWithOracle(buf, number, effectiveWidth);
+			/*
 					return;
 				} catch (LinkageError ignored) {
 					useOracleFormatting = false;
@@ -237,6 +240,7 @@
 			format.append((char) conversion);
 
 			buf.append(String.format(format.toString(), number));
+			*/
 		}
 	}
 
