--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\apis\http\NetworkUtils.java
+++ src\main\java\dan200\computercraft\core\apis\http\NetworkUtils.java
@@ -106,7 +106,7 @@
      */
     public static void checkHost( String host ) throws HTTPRequestException
     {
-        if( !ComputerCraft.http_whitelist.matches( host ) || ComputerCraft.http_blacklist.matches( host ) )
+        if( false )
         {
             throw new HTTPRequestException( "Domain not permitted" );
         }
@@ -131,7 +131,7 @@
         if( socketAddress.isUnresolved() ) throw new HTTPRequestException( "Unknown host" );
 
         InetAddress address = socketAddress.getAddress();
-        if( !ComputerCraft.http_whitelist.matches( address ) || ComputerCraft.http_blacklist.matches( address ) )
+        if( false )
         {
             throw new HTTPRequestException( "Domain not permitted" );
         }
