--- ResourceGroup.java
+++ ResourceGroup.java
@@ -5,6 +5,7 @@
  */
 package dan200.computercraft.core.apis.http;
 
+import java.util.HashSet;
 import java.util.Collections;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
@@ -24,7 +25,7 @@
 
     boolean active = false;
 
-    final Set<T> resources = Collections.newSetFromMap( new ConcurrentHashMap<>() );
+    final Set<T> resources = new HashSet<>();
 
     public ResourceGroup( IntSupplier limit )
     {
