--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\apis\http\ResourceGroup.java
+++ src\main\java\dan200\computercraft\core\apis\http\ResourceGroup.java
@@ -5,9 +5,8 @@
  */
 package dan200.computercraft.core.apis.http;
 
-import java.util.Collections;
+import java.util.HashSet;
 import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.IntSupplier;
 import java.util.function.Supplier;
 
@@ -24,7 +23,7 @@
 
     boolean active = false;
 
-    final Set<T> resources = Collections.newSetFromMap( new ConcurrentHashMap<>() );
+    final Set<T> resources = new HashSet<>();
 
     public ResourceGroup( IntSupplier limit )
     {
