--- original\CC-Tweaked\src\main\java\dan200\computercraft\api\peripheral\IComputerAccess.java
+++ src\main\java\dan200\computercraft\api\peripheral\IComputerAccess.java
@@ -6,12 +6,10 @@
 
 package dan200.computercraft.api.peripheral;
 
-import dan200.computercraft.api.ComputerCraftAPI;
 import dan200.computercraft.api.filesystem.IMount;
 import dan200.computercraft.api.filesystem.IWritableMount;
 import dan200.computercraft.api.lua.ILuaContext;
 import dan200.computercraft.api.lua.ILuaTask;
-import net.minecraft.world.World;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
@@ -33,8 +31,6 @@
      * @return The location on the computer's file system where you the mount mounted, or {@code null} if there was already a
      * file in the desired location. Store this value if you wish to unmount the mount later.
      * @throws RuntimeException If the peripheral has been detached.
-     * @see ComputerCraftAPI#createSaveDirMount(World, String, long)
-     * @see ComputerCraftAPI#createResourceMount(Class, String, String)
      * @see #mount(String, IMount, String)
      * @see #mountWritable(String, IWritableMount)
      * @see #unmount(String)
