--- LuaMethod.java
+++ LuaMethod.java
@@ -12,10 +12,6 @@
 
 public interface LuaMethod
 {
-    Generator<LuaMethod> GENERATOR = new Generator<>( LuaMethod.class, Collections.singletonList( ILuaContext.class ),
-        m -> ( target, context, args ) -> context.executeMainThreadTask( () -> ResultHelpers.checkNormalResult( m.apply( target, context, args ) ) )
-    );
-
     IntCache<LuaMethod> DYNAMIC = new IntCache<>(
         method -> ( instance, context, args ) -> ((IDynamicLuaObject) instance).callMethod( context, method, args )
     );
