--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\asm\PeripheralMethod.java
+++ src\main\java\dan200\computercraft\core\asm\PeripheralMethod.java
@@ -18,10 +18,6 @@
 
 public interface PeripheralMethod
 {
-    Generator<PeripheralMethod> GENERATOR = new Generator<>( PeripheralMethod.class, Arrays.asList( ILuaContext.class, IComputerAccess.class ),
-        m -> ( target, context, computer, args ) -> TaskCallback.make( context, () -> TaskCallback.checkUnwrap( m.apply( target, context, computer, args ) ) )
-    );
-
     IntCache<PeripheralMethod> DYNAMIC = new IntCache<>(
         method -> ( instance, context, computer, args ) -> ((IDynamicPeripheral) instance).callMethod( computer, context, method, args )
     );
