--- FSAPI.java
+++ FSAPI.java
@@ -20,9 +20,9 @@
 
 import java.io.BufferedReader;
 import java.io.BufferedWriter;
-import java.nio.channels.ReadableByteChannel;
-import java.nio.channels.WritableByteChannel;
-import java.nio.file.attribute.BasicFileAttributes;
+import cc.squiddev.cct.stub.ReadableByteChannel;
+import cc.squiddev.cct.stub.WritableByteChannel;
+import cc.squiddev.cct.stub.BasicFileAttributes;
 import java.nio.file.attribute.FileTime;
 import java.util.HashMap;
 import java.util.Map;
@@ -505,8 +505,8 @@
         }
     }
 
-    private static long getFileTime( FileTime time )
+    private static long getFileTime( long time )
     {
-        return time == null ? 0 : time.toMillis();
+        return time;
     }
 }
