--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\apis\HTTPAPI.java
+++ src\main\java\dan200\computercraft\core\apis\HTTPAPI.java
@@ -13,16 +13,10 @@
 import dan200.computercraft.core.apis.http.*;
 import dan200.computercraft.core.apis.http.request.HttpRequest;
 import dan200.computercraft.core.apis.http.websocket.Websocket;
-import io.netty.handler.codec.http.DefaultHttpHeaders;
-import io.netty.handler.codec.http.HttpHeaders;
-import io.netty.handler.codec.http.HttpMethod;
 
 import javax.annotation.Nonnull;
 import java.net.URI;
-import java.util.Collections;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
 
 import static dan200.computercraft.core.apis.TableHelper.*;
 
@@ -34,9 +28,10 @@
  */
 public class HTTPAPI implements ILuaAPI
 {
+    private static final String[] VERBS = new String[]{"DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT", "TRACE"};
+
     private final IAPIEnvironment m_apiEnvironment;
 
-    private final ResourceGroup<CheckUrl> checkUrls = new ResourceGroup<>();
     private final ResourceGroup<HttpRequest> requests = new ResourceQueue<>( () -> ComputerCraft.httpMaxRequests );
     private final ResourceGroup<Websocket> websockets = new ResourceGroup<>( () -> ComputerCraft.httpMaxWebsockets );
 
@@ -54,7 +49,6 @@
     @Override
     public void startup()
     {
-        checkUrls.startup();
         requests.startup();
         websockets.startup();
     }
@@ -62,7 +56,6 @@
     @Override
     public void shutdown()
     {
-        checkUrls.shutdown();
         requests.shutdown();
         websockets.shutdown();
     }
@@ -104,17 +97,17 @@
             redirect = true;
         }
 
-        HttpHeaders headers = getHeaders( headerTable );
+        Map<String, String> headers = getHeaders( headerTable );
 
-        HttpMethod httpMethod;
+        String httpMethod;
         if( requestMethod == null )
         {
-            httpMethod = postString == null ? HttpMethod.GET : HttpMethod.POST;
+            httpMethod = postString == null ? "GET" : "POST";
         }
         else
         {
-            httpMethod = HttpMethod.valueOf( requestMethod.toUpperCase( Locale.ROOT ) );
-            if( httpMethod == null || requestMethod.equalsIgnoreCase( "CONNECT" ) )
+            httpMethod = requestMethod.toUpperCase();
+            if( Arrays.binarySearch(VERBS, httpMethod) == -1 )
             {
                 throw new LuaException( "Unsupported HTTP method" );
             }
@@ -142,7 +135,8 @@
         try
         {
             URI uri = HttpRequest.checkUri( address );
-            new CheckUrl( checkUrls, m_apiEnvironment, address, uri ).queue( CheckUrl::run );
+            HttpRequest.checkUri( uri );
+            m_apiEnvironment.queueEvent("http_check", address, true);
 
             return new Object[] { true };
         }
@@ -160,7 +154,7 @@
             throw new LuaException( "Websocket connections are disabled" );
         }
 
-        HttpHeaders headers = getHeaders( headerTbl.orElse( Collections.emptyMap() ) );
+        Map<String, String> headers = getHeaders( headerTbl.orElse( Collections.emptyMap() ) );
 
         try
         {
@@ -179,9 +173,9 @@
     }
 
     @Nonnull
-    private static HttpHeaders getHeaders( @Nonnull Map<?, ?> headerTable ) throws LuaException
+    private static Map<String, String> getHeaders( @Nonnull Map<?, ?> headerTable ) throws LuaException
     {
-        HttpHeaders headers = new DefaultHttpHeaders();
+        Map<String, String> headers = new HashMap<>();
         for( Map.Entry<?, ?> entry : headerTable.entrySet() )
         {
             Object value = entry.getValue();
@@ -189,7 +183,7 @@
             {
                 try
                 {
-                    headers.add( (String) entry.getKey(), value );
+                    headers.put( ((String) entry.getKey()).toLowerCase(), (String)value );
                 }
                 catch( IllegalArgumentException e )
                 {
