--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\apis\OSAPI.java
+++ src\main\java\dan200\computercraft\core\apis\OSAPI.java
@@ -5,6 +5,7 @@
  */
 package dan200.computercraft.core.apis;
 
+import cc.squiddev.cct.stub.LuaDateTime;
 import dan200.computercraft.api.lua.IArguments;
 import dan200.computercraft.api.lua.ILuaAPI;
 import dan200.computercraft.api.lua.LuaException;
@@ -443,29 +444,24 @@
     public final Object date( Optional<String> formatA, Optional<Long> timeA ) throws LuaException
     {
         String format = formatA.orElse( "%c" );
-        long time = timeA.orElseGet( () -> Instant.now().getEpochSecond() );
+        long time = timeA.orElseGet( () -> Calendar.getInstance().getTimeInMillis() );
 
-        Instant instant = Instant.ofEpochSecond( time );
-        ZonedDateTime date;
-        ZoneOffset offset;
+        Calendar instant;
         if( format.startsWith( "!" ) )
         {
-            offset = ZoneOffset.UTC;
-            date = ZonedDateTime.ofInstant( instant, offset );
+            instant = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+            instant.setTimeInMillis(time);
             format = format.substring( 1 );
         }
         else
         {
-            ZoneId id = ZoneId.systemDefault();
-            offset = id.getRules().getOffset( instant );
-            date = ZonedDateTime.ofInstant( instant, id );
+            instant = Calendar.getInstance();
+            instant.setTimeInMillis(time);
         }
 
-        if( format.equals( "*t" ) ) return LuaDateTime.toTable( date, offset, instant );
+        if( format.equals( "*t" ) ) return LuaDateTime.toTable( instant );
 
-        DateTimeFormatterBuilder formatter = new DateTimeFormatterBuilder();
-        LuaDateTime.format( formatter, format, offset );
-        return formatter.toFormatter( Locale.ROOT ).format( date );
+        return instant.toString(); // TODO: Something!
     }
 
 }
