--- Palette.java
+++ Palette.java
@@ -5,10 +5,6 @@
  */
 package dan200.computercraft.shared.util;
 
-import dan200.computercraft.client.render.text.FixedWidthFontRenderer;
-import net.minecraft.nbt.CompoundNBT;
-import net.minecraft.network.PacketBuffer;
-
 import javax.annotation.Nonnull;
 
 public class Palette
@@ -101,6 +97,7 @@
         };
     }
 
+    /*
     public void write( PacketBuffer buffer )
     {
         for( double[] colour : colours )
@@ -146,4 +143,5 @@
             setColour( i, colours[0], colours[1], colours[2] );
         }
     }
+    */
 }
