--- original\CC-Tweaked\src\main\java\dan200\computercraft\shared\util\Palette.java
+++ src\main\java\dan200\computercraft\shared\util\Palette.java
@@ -5,8 +5,6 @@
  */
 package dan200.computercraft.shared.util;
 
-import net.minecraft.nbt.NBTTagCompound;
-
 public class Palette
 {
     private static final int PALETTE_SIZE = 16;
@@ -78,6 +76,7 @@
         };
     }
 
+    /*
     public NBTTagCompound writeToNBT( NBTTagCompound nbt )
     {
         int[] rgb8 = new int[colours.length];
@@ -103,4 +102,5 @@
             colours[i] = decodeRGB8( rgb8[i] );
         }
     }
+    */
 }
