--- original\CC-Tweaked\src\main\java\dan200\computercraft\shared\util\Palette.java
+++ src\main\java\dan200\computercraft\shared\util\Palette.java
@@ -6,8 +6,6 @@
 
 package dan200.computercraft.shared.util;
 
-import net.minecraft.nbt.NBTTagCompound;
-
 public class Palette
 {
     private static final int PALETTE_SIZE = 16;
@@ -79,6 +77,7 @@
         };
     }
 
+    /*
     public NBTTagCompound writeToNBT( NBTTagCompound nbt )
     {
         int[] rgb8 = new int[colours.length];
@@ -104,4 +103,5 @@
             colours[i] = decodeRGB8( rgb8[i] );
         }
     }
+    */
 }
