--- original\CC-Tweaked\src\main\java\dan200\computercraft\api\filesystem\IMount.java
+++ src\main\java\dan200\computercraft\api\filesystem\IMount.java
@@ -5,14 +5,12 @@
  */
 package dan200.computercraft.api.filesystem;
 
-import dan200.computercraft.api.ComputerCraftAPI;
 import dan200.computercraft.api.peripheral.IComputerAccess;
-import net.minecraft.world.World;
 
 import javax.annotation.Nonnull;
 import java.io.IOException;
-import java.nio.channels.ReadableByteChannel;
-import java.nio.file.attribute.BasicFileAttributes;
+import cc.squiddev.cct.stub.ReadableByteChannel;
+import cc.squiddev.cct.stub.BasicFileAttributes;
 import java.util.List;
 
 /**
@@ -89,6 +87,6 @@
     default BasicFileAttributes getAttributes( @Nonnull String path ) throws IOException
     {
         if( !exists( path ) ) throw new FileOperationException( path, "No such file" );
-        return new FileAttributes( isDirectory( path ), getSize( path ) );
+        return new BasicFileAttributes( isDirectory( path ), getSize( path ) );
     }
 }
