--- original\CC-Tweaked\src\main\java\dan200\computercraft\api\filesystem\IWritableMount.java
+++ src\main\java\dan200\computercraft\api\filesystem\IWritableMount.java
@@ -5,14 +5,12 @@
  */
 package dan200.computercraft.api.filesystem;
 
-import dan200.computercraft.api.ComputerCraftAPI;
 import dan200.computercraft.api.peripheral.IComputerAccess;
-import net.minecraft.world.World;
 
 import javax.annotation.Nonnull;
 import java.io.IOException;
 import java.io.OutputStream;
-import java.nio.channels.WritableByteChannel;
+import cc.squiddev.cct.stub.WritableByteChannel;
 import java.util.OptionalLong;
 
 /**
