--- UnsignedValues.java
+++ UnsignedValues.java
@@ -56,7 +56,7 @@
 
 	@SuppressWarnings("argument") // because API method isn't annotated
 	public static @Unsigned long toUlong(@Unsigned int value) {
-		return Integer.toUnsignedLong(value);
+		return ((long) value) & 0xffffffffL;
 	}
 
 	/**
@@ -64,7 +64,7 @@
 	 */
 	public static @Unsigned long toUlongFromSigned(@SignedPositive int value) {
 		if (value < 0) throw new IllegalArgumentException("value must be positive");
-		return Integer.toUnsignedLong(value);
+		return ((long) value) & 0xffffffffL;
 	}
 
 	public static @Unsigned int toUint(@Unsigned long value) {
