--- BinaryWritableHandle.java
+++ BinaryWritableHandle.java
@@ -14,8 +14,8 @@
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.nio.channels.FileChannel;
-import java.nio.channels.SeekableByteChannel;
-import java.nio.channels.WritableByteChannel;
+import cc.squiddev.cct.stub.SeekableByteChannel;
+import cc.squiddev.cct.stub.WritableByteChannel;
 import java.util.Optional;
 
 /**
@@ -96,6 +96,7 @@
     public final void flush() throws LuaException
     {
         checkOpen();
+        /*
         try
         {
             // Technically this is not needed
@@ -104,6 +105,7 @@
         catch( IOException ignored )
         {
         }
+        */
     }
 
     public static class Seekable extends BinaryWritableHandle
