--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\apis\handles\BinaryReadableHandle.java
+++ src\main\java\dan200\computercraft\core\apis\handles\BinaryReadableHandle.java
@@ -12,8 +12,8 @@
 import java.io.Closeable;
 import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.nio.channels.ReadableByteChannel;
-import java.nio.channels.SeekableByteChannel;
+import cc.squiddev.cct.stub.ReadableByteChannel;
+import cc.squiddev.cct.stub.SeekableByteChannel;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
