--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\apis\handles\EncodedReadableHandle.java
+++ src\main\java\dan200\computercraft\core\apis\handles\EncodedReadableHandle.java
@@ -12,8 +12,9 @@
 import java.io.BufferedReader;
 import java.io.Closeable;
 import java.io.IOException;
+import java.io.InputStreamReader;
 import java.nio.channels.Channels;
-import java.nio.channels.ReadableByteChannel;
+import cc.squiddev.cct.stub.ReadableByteChannel;
 import java.nio.charset.Charset;
 import java.nio.charset.CharsetDecoder;
 import java.nio.charset.CodingErrorAction;
@@ -173,9 +174,10 @@
 
     public static BufferedReader openUtf8( ReadableByteChannel channel )
     {
-        return open( channel, StandardCharsets.UTF_8 );
+        return new BufferedReader(new InputStreamReader(channel.asInputStream()));
     }
 
+    /*
     public static BufferedReader open( ReadableByteChannel channel, Charset charset )
     {
         // Create a charset decoder with the same properties as StreamDecoder does for
@@ -185,4 +187,5 @@
             .onUnmappableCharacter( CodingErrorAction.REPLACE );
         return new BufferedReader( Channels.newReader( channel, decoder, -1 ) );
     }
+    */
 }
