--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\terminal\Terminal.java
+++ src\main\java\dan200\computercraft\core\terminal\Terminal.java
@@ -6,7 +6,6 @@
 package dan200.computercraft.core.terminal;
 
 import dan200.computercraft.shared.util.Palette;
-import net.minecraft.nbt.NBTTagCompound;
 
 public class Terminal
 {
@@ -336,6 +335,7 @@
         m_changed = false;
     }
 
+    /*
     public synchronized NBTTagCompound writeToNBT( NBTTagCompound nbt )
     {
         nbt.setInteger( "term_cursorX", m_cursorX );
@@ -388,4 +388,5 @@
         }
         setChanged();
     }
+    */
 }
