--- Terminal.java
+++ Terminal.java
@@ -7,8 +7,6 @@
 
 import dan200.computercraft.shared.util.Colour;
 import dan200.computercraft.shared.util.Palette;
-import net.minecraft.nbt.CompoundNBT;
-import net.minecraft.network.PacketBuffer;
 
 import javax.annotation.Nonnull;
 
@@ -310,6 +308,7 @@
         if( onChanged != null ) onChanged.run();
     }
 
+    /*
     public synchronized void write( PacketBuffer buffer )
     {
         buffer.writeInt( cursorX );
@@ -421,4 +420,5 @@
         if( c >= 'a' && c <= 'f' ) return c - 'a' + 10;
         return 15 - def.ordinal();
     }
+    */
 }
