--- Terminal.java
+++ Terminal.java
@@ -7,8 +7,6 @@
 
 import dan200.computercraft.shared.util.Colour;
 import dan200.computercraft.shared.util.Palette;
-import net.minecraft.nbt.CompoundNBT;
-import net.minecraft.network.PacketBuffer;
 
 import javax.annotation.Nonnull;
 import java.nio.ByteBuffer;
@@ -311,6 +309,7 @@
         if( onChanged != null ) onChanged.run();
     }
 
+    /*
     public synchronized void write( PacketBuffer buffer )
     {
         buffer.writeInt( cursorX );
@@ -422,4 +421,5 @@
         if( c >= 'A' && c <= 'F' ) return c - 'A' + 10;
         return 15 - def.ordinal();
     }
+    */
 }
