--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\terminal\Terminal.java
+++ src\main\java\dan200\computercraft\core\terminal\Terminal.java
@@ -7,8 +7,6 @@
 
 import dan200.computercraft.shared.util.Colour;
 import dan200.computercraft.shared.util.Palette;
-import net.minecraft.nbt.NBTTagCompound;
-import net.minecraft.network.PacketBuffer;
 
 public class Terminal
 {
@@ -327,6 +325,7 @@
         m_changed = false;
     }
 
+    /*
     public synchronized void write( PacketBuffer buffer )
     {
         buffer.writeInt( m_cursorX );
@@ -431,6 +430,7 @@
         m_palette.readFromNBT( nbt );
         setChanged();
     }
+    */
 
     public static int getColour( char c, Colour def )
     {
