--- TimeoutState.java
+++ TimeoutState.java
@@ -48,7 +48,7 @@
      */
     public static final String ABORT_MESSAGE = "Too long without yielding";
 
-    private final ComputerThread scheduler;
+    private final cc.tweaked.web.ComputerThread scheduler;
 
     private boolean paused;
     private boolean softAbort;
@@ -74,7 +74,7 @@
      */
     private long currentDeadline;
 
-    public TimeoutState(ComputerThread scheduler) {
+    public TimeoutState(cc.tweaked.web.ComputerThread scheduler) {
         this.scheduler = scheduler;
     }
 
