--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\filesystem\EmptyMount.java
+++ src\main\java\dan200\computercraft\core\filesystem\EmptyMount.java
@@ -6,12 +6,11 @@
 
 package dan200.computercraft.core.filesystem;
 
+import cc.squiddev.cct.stub.ReadableByteChannel;
 import dan200.computercraft.api.filesystem.IMount;
 
 import javax.annotation.Nonnull;
 import java.io.IOException;
-import java.io.InputStream;
-import java.nio.channels.ReadableByteChannel;
 import java.util.List;
 
 public class EmptyMount implements IMount
@@ -42,16 +41,8 @@
     @Nonnull
     @Override
     @Deprecated
-    public InputStream openForRead( @Nonnull String path ) throws IOException
+    public ReadableByteChannel openChannelForRead(@Nonnull String path ) throws IOException
     {
         throw new IOException( "/" + path + ": No such file" );
     }
-
-    @Nonnull
-    @Override
-    @Deprecated
-    public ReadableByteChannel openChannelForRead( @Nonnull String path ) throws IOException
-    {
-        throw new IOException( "/" + path + ": No such file" );
-    }
 }
