--- FileSystem.java
+++ FileSystem.java
@@ -422,7 +422,7 @@
         // Collapse the string into its component parts, removing ..'s
         var outputParts = new ArrayDeque<String>();
         for (var fullPart : Splitter.on('/').split(path)) {
-            var part = fullPart.strip();
+            var part = fullPart.trim();
 
             if (part.isEmpty() || part.equals(".") || threeDotsPattern.matcher(part).matches()) {
                 // . is redundant
@@ -444,7 +444,7 @@
                 }
             } else if (part.length() >= 255) {
                 // If part length > 255 and it is the last part
-                outputParts.addLast(part.substring(0, 255).strip());
+                outputParts.addLast(part.substring(0, 255).trim());
             } else {
                 // Anything else we add to the stack
                 outputParts.addLast(part);
