--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\filesystem\FileSystemWrapperMount.java
+++ src\main\java\dan200\computercraft\core\filesystem\FileSystemWrapperMount.java
@@ -9,8 +9,8 @@
 
 import javax.annotation.Nonnull;
 import java.io.IOException;
-import java.nio.channels.ReadableByteChannel;
-import java.nio.channels.WritableByteChannel;
+import cc.squiddev.cct.stub.ReadableByteChannel;
+import cc.squiddev.cct.stub.WritableByteChannel;
 import java.util.Collections;
 import java.util.List;
 import java.util.function.Function;
