--- MountWrapper.java
+++ MountWrapper.java
@@ -217,7 +217,7 @@
             if (ex.getFilename() != null) return localExceptionOf(ex.getFilename(), FileSystemException.getMessage(ex));
         }
 
-        if (e instanceof java.nio.file.FileSystemException ex) {
+        if (e instanceof cc.tweaked.web.stub.FileSystemException ex) {
             // This error will contain the absolute path, leaking information about where MC is installed. We drop that,
             // just taking the reason. We assume that the error refers to the input path.
             var message = ex.getReason();
