--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\filesystem\MountWrapper.java
+++ src\main\java\dan200\computercraft\core\filesystem\MountWrapper.java
@@ -12,10 +12,10 @@
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import java.io.IOException;
-import java.nio.channels.ReadableByteChannel;
-import java.nio.channels.WritableByteChannel;
+import cc.squiddev.cct.stub.ReadableByteChannel;
+import cc.squiddev.cct.stub.WritableByteChannel;
 import java.nio.file.AccessDeniedException;
-import java.nio.file.attribute.BasicFileAttributes;
+import cc.squiddev.cct.stub.BasicFileAttributes;
 import java.util.List;
 import java.util.OptionalLong;
 
@@ -204,10 +204,6 @@
                 writableMount.delete( path );
             }
         }
-        catch( AccessDeniedException e )
-        {
-            throw new FileSystemException( "Access denied" );
-        }
         catch( IOException e )
         {
             throw localExceptionOf( path, e );
@@ -238,10 +234,6 @@
                 return writableMount.openForWrite( path );
             }
         }
-        catch( AccessDeniedException e )
-        {
-            throw new FileSystemException( "Access denied" );
-        }
         catch( IOException e )
         {
             throw localExceptionOf( path, e );
@@ -276,10 +268,6 @@
                 return writableMount.openForAppend( path );
             }
         }
-        catch( AccessDeniedException e )
-        {
-            throw new FileSystemException( "Access denied" );
-        }
         catch( IOException e )
         {
             throw localExceptionOf( path, e );
@@ -299,6 +287,7 @@
             if( ex.getFilename() != null ) return localExceptionOf( ex.getFilename(), ex.getMessage() );
         }
 
+        /*
         if( e instanceof java.nio.file.FileSystemException )
         {
             // This error will contain the absolute path, leaking information about where MC is installed. We drop that,
@@ -306,6 +295,7 @@
             String message = ((java.nio.file.FileSystemException) e).getReason().trim();
             return localPath == null ? new FileSystemException( message ) : localExceptionOf( localPath, message );
         }
+        */
 
         return new FileSystemException( e.getMessage() );
     }
