--- LuaThread.java
+++ LuaThread.java
@@ -30,11 +30,10 @@
 import org.squiddev.cobalt.function.LuaFunction;
 import org.squiddev.cobalt.lib.CoroutineLib;
 import org.squiddev.cobalt.lib.jse.JsePlatform;
+import org.teavm.interop.Sync;
 
 import java.lang.ref.WeakReference;
 import java.util.Objects;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.locks.Condition;
 
 import static org.squiddev.cobalt.debug.DebugFrame.FLAG_ERROR;
 import static org.squiddev.cobalt.debug.DebugFrame.FLAG_YPCALL;
@@ -60,12 +59,6 @@
  */
 public class LuaThread extends LuaValue {
 	/**
-	 * Interval in nanoseconds at which to check for lua threads that are no longer referenced.
-	 * This can be changed by Java startup code if desired.
-	 */
-	public static long orphanCheckInterval = TimeUnit.SECONDS.toNanos(30);
-
-	/**
 	 * A coroutine which has been run at all
 	 */
 	private static final int STATUS_INITIAL = 0;
@@ -262,14 +255,11 @@
 		if (state.currentThread.state.javaCount == 0) {
 			throw UnwindThrowable.yield(args);
 		} else {
-			try {
-				return yieldBlockingImpl(state, args);
-			} catch (InterruptedException e) {
-				throw new InterruptedError(e);
-			}
+			throw new LuaError("attempt to yield across a call boundary");
 		}
 	}
 
+	/*
 	/**
 	 * Yield the current thread and wait for a response
 	 *
@@ -278,12 +268,12 @@
 	 * @return The values this coroutine was resumed with
 	 * @throws LuaError             If this thread cannot be yielded.
 	 * @throws InterruptedException If this thread was terminated when yielding.
-	 */
 	public static Varargs yieldBlocking(LuaState state, Varargs args) throws LuaError, InterruptedException {
 		Objects.requireNonNull(args, "args cannot be null");
 		checkYield(state);
 		return yieldBlockingImpl(state, args);
 	}
+	*/
 
 	private static void checkYield(LuaState state) throws LuaError {
 		LuaThread thread = state.currentThread;
@@ -293,6 +283,7 @@
 		if (thread.isMainThread()) throw new LuaError("cannot yield main thread");
 	}
 
+	/*
 	private static Varargs yieldBlockingImpl(LuaState state, Varargs args) throws InterruptedException, LuaError {
 		State current = state.currentThread.state;
 
@@ -302,6 +293,7 @@
 		current.previousThread = null;
 		return transferControl(state, current, args);
 	}
+	*/
 
 	/**
 	 * Resume a thread with arguments.
@@ -328,6 +320,7 @@
 		if (currentState.javaCount == 0) {
 			throw UnwindThrowable.resume(thread, args);
 		} else {
+			/*
 			try {
 				// Mark the child coroutine as "active", and transfer.
 				state.currentThread = thread;
@@ -342,9 +335,12 @@
 			} catch (InterruptedException e) {
 				throw new InterruptedError(e);
 			}
+			*/
+			throw new LuaError("attempt to resume across a call boundary");
 		}
 	}
 
+	/*
 	/**
 	 * Suspend the current thread. Note, this may return or throw an exception, so you must handle both cases.
 	 *
@@ -361,7 +357,7 @@
 		if (current.javaCount == 0) {
 			throw UnwindThrowable.suspend();
 		} else {
-			suspendBlocking(state);
+			throw new LuaError("attempt to yield across a call boundary");
 		}
 	}
 
@@ -370,7 +366,6 @@
 	 *
 	 * @param state The current lua state
 	 * @throws LuaError If this coroutine cannot be suspended.
-	 */
 	public static void suspendBlocking(LuaState state) throws LuaError {
 		State current = state.currentThread.state;
 		if (current.status != STATUS_RUNNING) {
@@ -412,6 +407,7 @@
 			thread.needsThreadedResume = false;
 		}
 	}
+	*/
 
 	public static Varargs runMain(LuaState state, LuaFunction function) throws LuaError, InterruptedException {
 		return run(state, state.getMainThread(), function, Constants.NONE);
@@ -434,6 +430,7 @@
 	}
 
 	private static Varargs run(final LuaState state, LuaThread thread, LuaFunction function, Varargs args) throws LuaError, InterruptedException {
+		/*
 		YieldThreader threader = state.threader;
 		threader.lock.lock();
 		try {
@@ -491,11 +488,12 @@
 		} finally {
 			threader.lock.unlock();
 		}
+		*/
+		return loop(state, thread, function, args);
 	}
 
-	static Varargs loop(final LuaState state, LuaThread thread, LuaFunction function, Varargs args) throws LuaError, TransferredControlThrowable {
-		YieldThreader threader = state.threader;
-
+	@Sync
+	static Varargs loop(final LuaState state, LuaThread thread, LuaFunction function, Varargs args) throws LuaError {
 		LuaError le = null;
 		do {
 			final State threadState = thread.state;
@@ -521,6 +519,7 @@
 						args = null;
 						le = LuaError.wrap(e);
 					}
+				/*
 				} else if (threadState.needsThreadedResume) {
 					// We only ever resume coroutines which have yielded, never those which have
 					// resumed other coroutines. Consequently, we know we will never have an error here.
@@ -534,6 +533,7 @@
 					threadState.resumeLock.signal();
 
 					throw TransferredControlThrowable.INSTANCE;
+				*/
 				} else {
 					threadState.status = STATUS_RUNNING;
 
@@ -656,16 +656,6 @@
 		int javaCount = 0;
 
 		/**
-		 * The lock to wait on while this coroutine is suspended as a thread
-		 */
-		Condition resumeLock;
-
-		/**
-		 * Whether we've yielded in a threaded manner.
-		 */
-		boolean needsThreadedResume;
-
-		/**
 		 * Constructor for main thread only
 		 */
 		State(LuaThread owner, int status) {
