--- CobaltLuaMachine.java
+++ CobaltLuaMachine.java
@@ -41,12 +41,14 @@
 
 public class CobaltLuaMachine implements ILuaMachine
 {
+    /*
     private static final ThreadPoolExecutor COROUTINES = new ThreadPoolExecutor(
         0, Integer.MAX_VALUE,
         5L, TimeUnit.MINUTES,
         new SynchronousQueue<>(),
         ThreadUtils.factory( "Coroutine" )
     );
+    */
 
     private static final LuaMethod FUNCTION_METHOD = ( target, context, args ) -> ((ILuaFunction) target).call( args );
 
@@ -72,6 +74,7 @@
         LuaState state = this.state = LuaState.builder()
             .resourceManipulator( new VoidResourceManipulator() )
             .debug( debug )
+            /*
             .coroutineExecutor( command -> {
                 Tracking.addValue( this.computer, TrackingField.COROUTINES_CREATED, 1 );
                 COROUTINES.execute( () -> {
@@ -85,6 +88,7 @@
                     }
                 } );
             } )
+            */
             .build();
 
         globals = new LuaTable();
@@ -238,7 +242,7 @@
             table.rawset( method, new ResultInterpreterFunction( this, LuaMethod.DYNAMIC.get( i ), object, context, method ) );
         }
 
-        ObjectSource.allMethods( LuaMethod.GENERATOR, object, ( instance, method ) ->
+        ObjectSource.allMethods( object, ( instance, method ) ->
             table.rawset( method.getName(), method.nonYielding()
                 ? new BasicFunction( this, method.getMethod(), instance, context, method.getName() )
                 : new ResultInterpreterFunction( this, method.getMethod(), instance, context, method.getName() ) ) );
@@ -455,7 +459,7 @@
             if( isPaused ) resetPaused( ds, di );
 
             // We check our current pause/abort state every 128 instructions.
-            if( (count = (count + 1) & 127) == 0 )
+            if( (count = (count + 1) & 1023) == 0 )
             {
                 // If we've been hard aborted or closed then abort.
                 if( timeout.isHardAborted() || state == null ) throw HardAbortError.INSTANCE;
@@ -488,7 +492,6 @@
             if( timeout.isHardAborted() || state == null ) throw HardAbortError.INSTANCE;
 
             timeout.refresh();
-            if( timeout.isPaused() ) LuaThread.suspendBlocking( state );
             handleSoftAbort();
         }
 
