--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\lua\CobaltLuaMachine.java
+++ src\main\java\dan200\computercraft\core\lua\CobaltLuaMachine.java
@@ -41,12 +41,14 @@
 
 public class CobaltLuaMachine implements ILuaMachine
 {
+    /*
     private static final ThreadPoolExecutor COROUTINES = new ThreadPoolExecutor(
         0, Integer.MAX_VALUE,
         5L, TimeUnit.MINUTES,
         new SynchronousQueue<>(),
         ThreadUtils.factory( "Coroutine" )
     );
+    */
 
     private static final LuaMethod FUNCTION_METHOD = ( target, context, args ) -> ((ILuaFunction) target).call( args );
 
@@ -71,6 +73,7 @@
         LuaState state = m_state = LuaState.builder()
             .resourceManipulator( new VoidResourceManipulator() )
             .debug( debug )
+            /*
             .coroutineExecutor( command -> {
                 Tracking.addValue( m_computer, TrackingField.COROUTINES_CREATED, 1 );
                 COROUTINES.execute( () -> {
@@ -84,6 +87,7 @@
                     }
                 } );
             } )
+            */
             .build();
 
         m_globals = new LuaTable();
@@ -122,7 +126,7 @@
         LuaTable table = wrapLuaObject( api );
         if( table == null )
         {
-            ComputerCraft.log.warn( "API {} does not provide any methods", api );
+            ComputerCraft.log.warn( "API does not provide any methods" );
             table = new LuaTable();
         }
 
@@ -237,7 +241,7 @@
             table.rawset( method, new ResultInterpreterFunction( this, LuaMethod.DYNAMIC.get( i ), object, context, method ) );
         }
 
-        ObjectSource.allMethods( LuaMethod.GENERATOR, object, ( instance, method ) ->
+        ObjectSource.allMethods( object, ( instance, method ) ->
             table.rawset( method.getName(), method.nonYielding()
                 ? new BasicFunction( this, method.getMethod(), instance, context, method.getName() )
                 : new ResultInterpreterFunction( this, method.getMethod(), instance, context, method.getName() ) ) );
@@ -332,7 +336,7 @@
 
         if( ComputerCraft.logComputerErrors )
         {
-            ComputerCraft.log.warn( "Received unknown type '{}', returning nil.", object.getClass().getName() );
+            ComputerCraft.log.warn( "Received unknown type returning nil." );
         }
         return Constants.NIL;
     }
@@ -487,7 +491,6 @@
             if( timeout.isHardAborted() || state == null ) throw HardAbortError.INSTANCE;
 
             timeout.refresh();
-            if( timeout.isPaused() ) LuaThread.suspendBlocking( state );
             handleSoftAbort();
         }
 
@@ -514,6 +517,8 @@
         @Override
         public long issueMainThreadTask( @Nonnull final ILuaTask task ) throws LuaException
         {
+            throw new LuaException("Cannot throw main-thread task");
+            /*
             // Issue command
             final long taskID = MainThread.getUniqueTaskID();
             final Runnable iTask = () -> {
@@ -553,6 +558,7 @@
             {
                 throw new LuaException( "Task limit exceeded" );
             }
+            */
         }
     }
 
