--- LuaContext.java
+++ LuaContext.java
@@ -27,6 +27,7 @@
     public long issueMainThreadTask( @Nonnull final ILuaTask task ) throws LuaException
     {
         // Issue command
+        /*
         final long taskID = MainThread.getUniqueTaskID();
         final Runnable iTask = () -> {
             try
@@ -65,5 +66,7 @@
         {
             throw new LuaException( "Task limit exceeded" );
         }
+        */
+        throw new LuaException("Cannot throw main-thread task");
     }
 }
