--- LuaContext.java
+++ LuaContext.java
@@ -29,6 +29,7 @@
     public long issueMainThreadTask( @Nonnull final ILuaTask task ) throws LuaException
     {
         // Issue command
+        /*
         final long taskID = MainThread.getUniqueTaskID();
         final Runnable iTask = () -> {
             try
@@ -67,6 +68,8 @@
         {
             throw new LuaException( "Task limit exceeded" );
         }
+        */
+        throw new LuaException("Cannot throw main-thread task");
     }
 
     @Nonnull
