--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\tracking\TrackingContext.java
+++ src\main\java\dan200\computercraft\core\tracking\TrackingContext.java
@@ -28,7 +28,6 @@
 
     public synchronized void start()
     {
-        if( !tracking ) Tracking.tracking.incrementAndGet();
         tracking = true;
 
         timings.clear();
@@ -39,7 +38,6 @@
     {
         if( !tracking ) return false;
 
-        Tracking.tracking.decrementAndGet();
         tracking = false;
         timingLookup.clear();
         return true;
