--- Tracking.java
+++ Tracking.java
@@ -15,6 +15,7 @@
 
 public final class Tracking
 {
+    /*
     static final AtomicInteger tracking = new AtomicInteger( 0 );
 
     private static final Object lock = new Object();
@@ -41,9 +42,11 @@
             tracking.incrementAndGet();
         }
     }
+    */
 
     public static void addTaskTiming( Computer computer, long time )
     {
+        /*
         if( tracking.get() == 0 ) return;
 
         synchronized( contexts )
@@ -51,10 +54,12 @@
             for( TrackingContext context : contexts.values() ) context.addTaskTiming( computer, time );
             for( Tracker tracker : trackers ) tracker.addTaskTiming( computer, time );
         }
+        */
     }
 
     public static void addServerTiming( Computer computer, long time )
     {
+        /*
         if( tracking.get() == 0 ) return;
 
         synchronized( contexts )
@@ -62,10 +67,12 @@
             for( TrackingContext context : contexts.values() ) context.addServerTiming( computer, time );
             for( Tracker tracker : trackers ) tracker.addServerTiming( computer, time );
         }
+        */
     }
 
     public static void addValue( Computer computer, TrackingField field, long change )
     {
+        /*
         if( tracking.get() == 0 ) return;
 
         synchronized( lock )
@@ -73,8 +80,10 @@
             for( TrackingContext context : contexts.values() ) context.addValue( computer, field, change );
             for( Tracker tracker : trackers ) tracker.addValue( computer, field, change );
         }
+        */
     }
 
+    /*
     public static void reset()
     {
         synchronized( lock )
@@ -84,4 +93,5 @@
             tracking.set( 0 );
         }
     }
+    */
 }
