--- original\CC-Tweaked\src\main\java\dan200\computercraft\core\tracking\TrackingField.java
+++ src\main\java\dan200\computercraft\core\tracking\TrackingField.java
@@ -8,7 +8,7 @@
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.function.LongFunction;
+import java.util.function.Function;
 
 public final class TrackingField
 {
@@ -38,7 +38,7 @@
 
     private final String id;
     private final String translationKey;
-    private final LongFunction<String> format;
+    private final Function<Long, String> format;
 
     public String id()
     {
@@ -50,7 +50,7 @@
         return translationKey;
     }
 
-    private TrackingField( String id, LongFunction<String> format )
+    private TrackingField( String id, Function<Long, String> format )
     {
         this.id = id;
         translationKey = "tracking_field.computercraft." + id + ".name";
@@ -62,7 +62,7 @@
         return format.apply( value );
     }
 
-    public static TrackingField of( String id, LongFunction<String> format )
+    public static TrackingField of( String id, Function<Long, String> format )
     {
         TrackingField field = new TrackingField( id, format );
         fields.put( id, field );
