--- ComputerCraft.java
+++ ComputerCraft.java
@@ -5,6 +5,7 @@
  */
 package dan200.computercraft;
 
+/*
 import dan200.computercraft.api.turtle.event.TurtleAction;
 import dan200.computercraft.core.apis.http.options.Action;
 import dan200.computercraft.core.apis.http.options.AddressRule;
@@ -19,14 +20,14 @@
 import net.minecraftforge.fml.common.Mod;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-
+*/
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.EnumSet;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
+import cc.squiddev.cct.stub.Logger;
 
-@Mod( ComputerCraft.MOD_ID )
 public final class ComputerCraft
 {
     public static final String MOD_ID = "computercraft";
@@ -40,15 +41,17 @@
     public static boolean commandRequireCreative = true;
 
     public static int computerThreads = 1;
-    public static long maxMainGlobalTime = TimeUnit.MILLISECONDS.toNanos( 10 );
-    public static long maxMainComputerTime = TimeUnit.MILLISECONDS.toNanos( 5 );
+    public static long maxMainGlobalTime = 10 * 1_000_000L;
+    public static long maxMainComputerTime = 5 * 1_000_000L;
 
     public static boolean httpEnabled = true;
     public static boolean httpWebsocketEnabled = true;
+    /*
     public static List<AddressRule> httpRules = Collections.unmodifiableList( Arrays.asList(
         AddressRule.parse( "$private", null, Action.DENY.toPartial() ),
         AddressRule.parse( "*", null, Action.ALLOW.toPartial() )
     ) );
+    */
 
     public static int httpMaxRequests = 16;
     public static int httpMaxWebsockets = 4;
@@ -61,7 +64,6 @@
     public static int modemRangeDuringStorm = 64;
     public static int modemHighAltitudeRangeDuringStorm = 384;
     public static int maxNotesPerTick = 8;
-    public static MonitorRenderer monitorRenderer = MonitorRenderer.BEST;
     public static double monitorDistanceSq = 4096;
     public static long monitorBandwidth = 1_000_000;
 
@@ -70,7 +72,6 @@
     public static int advancedTurtleFuelLimit = 100000;
     public static boolean turtlesObeyBlockProtection = true;
     public static boolean turtlesCanPush = true;
-    public static EnumSet<TurtleAction> turtleDisabledActions = EnumSet.noneOf( TurtleAction.class );
 
     public static int computerTermWidth = 51;
     public static int computerTermHeight = 19;
@@ -84,6 +85,14 @@
     public static int monitorWidth = 8;
     public static int monitorHeight = 6;
 
+    public static final Logger log = Logger.INSTANCE;
+    static final String VERSION = "1.100.0";
+
+    public static String getVersion() {
+        return VERSION;
+    }
+
+    /*
     public static final class TurtleUpgrades
     {
         public static TurtleModem wirelessModemNormal;
@@ -117,4 +126,5 @@
         Config.setup();
         Registry.setup();
     }
+    */
 }
