--- original\CC-Tweaked\src\main\java\dan200\computercraft\api\lua\ILuaObject.java
+++ src\main\java\dan200\computercraft\api\lua\ILuaObject.java
@@ -7,6 +7,7 @@
 
 import dan200.computercraft.api.peripheral.IComputerAccess;
 import dan200.computercraft.api.peripheral.IPeripheral;
+import org.squiddev.cobalt.LuaTable;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
@@ -52,4 +53,7 @@
      */
     @Nullable
     Object[] callMethod( @Nonnull ILuaContext context, int method, @Nonnull Object[] arguments ) throws LuaException, InterruptedException;
+
+    default void overwriteLuaTable(LuaTable table) {
+    }
 }
