--- Resource.java
+++ Resource.java
@@ -6,7 +6,6 @@
 package dan200.computercraft.core.apis.http;
 
 import dan200.computercraft.core.util.IoUtil;
-import io.netty.channel.ChannelFuture;
 
 import javax.annotation.Nullable;
 import java.io.Closeable;
@@ -101,6 +100,7 @@
         return null;
     }
 
+    /*
     @Nullable
     protected static ChannelFuture closeChannel(@Nullable ChannelFuture future) {
         if (future != null) {
@@ -111,6 +111,7 @@
 
         return null;
     }
+    */
 
     @Nullable
     protected static <T extends Future<?>> T closeFuture(@Nullable T future) {
