--- Resource.java
+++ Resource.java
@@ -5,7 +5,6 @@
 package dan200.computercraft.core.apis.http;
 
 import dan200.computercraft.core.util.IoUtil;
-import io.netty.channel.ChannelFuture;
 
 import javax.annotation.Nullable;
 import java.io.Closeable;
@@ -100,6 +99,7 @@
         return null;
     }
 
+    /*
     @Nullable
     protected static ChannelFuture closeChannel(@Nullable ChannelFuture future) {
         if (future != null) {
@@ -110,6 +110,7 @@
 
         return null;
     }
+    */
 
     @Nullable
     protected static <T extends Future<?>> T closeFuture(@Nullable T future) {
