--- ObjectSource.java
+++ ObjectSource.java
@@ -16,12 +16,12 @@
 public interface ObjectSource {
     Iterable<Object> getExtra();
 
-    static <T> void allMethods(Generator<T> generator, Object object, BiConsumer<Object, NamedMethod<T>> accept) {
-        for (var method : generator.getMethods(object.getClass())) accept.accept(object, method);
+    static void allMethods(Object object, BiConsumer<Object, NamedMethod<LuaMethod>> accept) {
+        for (var method : JsGenerator.getMethods(object.getClass())) accept.accept(object, method);
 
         if (object instanceof ObjectSource source) {
             for (var extra : source.getExtra()) {
-                for (var method : generator.getMethods(extra.getClass())) accept.accept(extra, method);
+                for (var method : JsGenerator.getMethods(extra.getClass())) accept.accept(extra, method);
             }
         }
     }
